app/src/main/java/com/allan/boardbuddies/fragments/NotesFragment.java

```diff
@@ -63,23 +70,57 @@ public void onViewCreated (View view, Bundle savedInstanceState){
         FloatingActionButton addNoteFab = view.findViewById(R.id.add_note_fab);
         addNoteFab.setOnClickListener(v -> {
             Intent intent = new Intent(v.getContext(), EditActivity.class);
-            v.getContext().startActivity(intent);
+            mNoteLauncher.launch(intent);
         });
     }
-
-    @Override
-    public void onResume() {
-        super.onResume();
-        loadNotes();
-        adapter.notifyDataSetChanged();
-    }
+    
+    ActivityResultLauncher<Intent> mNoteLauncher = registerForActivityResult(new StartActivityForResult(),
+            new ActivityResultCallback<ActivityResult>() {
+                @Override
+                public void onActivityResult(ActivityResult result) {
+                    if (result.getResultCode() == Activity.RESULT_OK) {
+                        Intent data = result.getData();
+                        String addedFilename = data.getStringExtra("addedFilename");
+                        int deletedPosition = data.getIntExtra("deletedPosition", -1);
+                        if (addedFilename != null){
+                            loadSingle(addedFilename, 0);
+                            adapter.notifyItemInserted(0);
+                        }
+                        if (deletedPosition != -1){
+                            notes.remove(deletedPosition);
+                            adapter.notifyItemRemoved(deletedPosition+1);
+                        }
+                    }
+                }
+            });
 
     public void onNoteClick(int position){
         Intent intent = new Intent(requireContext(), EditActivity.class);
         intent.putExtra("TITLE", notes.get(position).getTitle());
         intent.putExtra("CONTENT", notes.get(position).getContent());
         intent.putExtra("FILENAME", notes.get(position).getName());
-        requireContext().startActivity(intent);
+        intent.putExtra("POSITION", position);
+        mNoteLauncher.launch(intent);
+    }
+
+    private void loadSingle(String filename, int position){
+        File file = new File(getActivity().getApplicationContext().getFilesDir(), filename);
+        try {
```

Comment:
can do in follow-up: you can probably extract most of this into a helper since it's duplicated below in loadNotes